From 995a1063dbdc94e2c0846216363636e8ebbc60f9 Mon Sep 17 00:00:00 2001
From: Daniel Mack <daniel@zonque.org>
Date: Thu, 2 Oct 2014 17:12:13 +0200
Subject: [PATCH] sd-bus: sync kdbus.h (ABI break!)

struct kdbus_cmd_match got a flags field, which systemd currently makes no
use of.
---
 src/libsystemd/sd-bus/kdbus.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/libsystemd/sd-bus/kdbus.h b/src/libsystemd/sd-bus/kdbus.h
index c97994974b..801125946b 100644
--- a/src/libsystemd/sd-bus/kdbus.h
+++ b/src/libsystemd/sd-bus/kdbus.h
@@ -714,6 +714,16 @@ struct kdbus_cmd_update {
 } __attribute__((aligned(8)));
 
 /**
+ * enum kdbus_cmd_match_flags - flags to control the KDBUS_CMD_MATCH_ADD ioctl
+ * @KDBUS_MATCH_REPLACE:	If entries with the supplied cookie already
+ *				exists, remove them before installing the new
+ *				matches.
+ */
+enum kdbus_cmd_match_flags {
+	KDBUS_MATCH_REPLACE	= 1ULL <<  0,
+};
+
+/**
  * struct kdbus_cmd_match - struct to add or remove matches
  * @size:		The total size of the struct
  * @cookie:		Userspace supplied cookie. When removing, the cookie
@@ -726,6 +736,7 @@ struct kdbus_cmd_update {
 struct kdbus_cmd_match {
 	__u64 size;
 	__u64 cookie;
+	__u64 flags;
 	struct kdbus_item items[0];
 } __attribute__((aligned(8)));
 
